common/test/acceptance/tests/discussion/test_discussion.py

```diff
@@ -1262,6 +1270,43 @@ def test_redirects_to_learner_profile(self):
         learner_profile_page.wait_for_page()
         self.assertTrue(learner_profile_page.field_is_visible('username'))
 
+    def test_learner_profile_roles(self):
+        """
+        Test that on the learner profile page user roles are correctly listed according to the course.
+        """
+        # Setup a learner with roles in a Course-A.
+        expected_student_roles = ['Administrator', 'Community TA', 'Moderator', 'Student']
+        self.profiled_user_id = self.setup_user(
+            roles=expected_student_roles,
+            username=self.PROFILED_USERNAME
+        )
+
+        # Visit the page and verify the roles are listed correctly.
+        page = self.check_pages(1)
+        student_roles = page.get_user_roles()
+        self.assertTrue(
+            all([role in student_roles for role in expected_student_roles])
```

Comment:
btw, i have changed the `assert` to be much simpler now. thank you for brining it up.